src/vcs/git.rs

```diff
@@ -192,6 +204,65 @@ impl<'repo> Repository {
         self.commit_to_history(commit)
     }
 
+    /// Get the [`Diff`] between two commits.
+    pub fn diff(&self, from: &'repo git2::Commit, to: &'repo git2::Commit) -> Result<Diff, Error> {
+        use git2::{Delta, Patch};
+
+        let mut diff = Diff::new();
+        let git_diff = self.diff_commits(from, Some(to))?;
+
+        for (idx, delta) in git_diff.deltas().enumerate() {
+            match delta.status() {
+                Delta::Added => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
+
+                    diff.add_created_file(path);
+                },
+                Delta::Deleted => {
+                    let diff_file = delta.old_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
+
+                    diff.add_deleted_file(path);
+                },
+                Delta::Modified => {
+                    let diff_file = delta.new_file();
+                    let path = diff_file.path().expect("file should have a path");
+                    let path = file_system::Path::try_from(path.to_path_buf()).unwrap();
+
+                    let patch = Patch::from_diff(&git_diff, idx)?;
+
+                    if let Some(patch) = patch {
+                        let mut hunks: Vec<Hunk> = Vec::new();
+
+                        for h in 0..patch.num_hunks() {
+                            let (hunk, hunk_lines) = patch.hunk(h)?;
+                            let header = hunk.header().to_owned();
+                            let mut lines: Vec<LineDiff> = Vec::new();
+
+                            for l in 0..hunk_lines {
+                                let line = patch.line_in_hunk(h, l)?;
+                                lines.push(line.into());
+                            }
+                            hunks.push(Hunk { header, lines });
+                        }
+                        diff.add_modified_file(path, hunks);
+                    } else if diff_file.is_binary() {
+                        // TODO
+                    } else {
+                        unreachable!()
+                    }
+                },
+                _ => {},
+            }
+            //
```

Comment:
?

Reply:
These ones seem reasonable to handle:
```
    Modified,
    Renamed,
    Copied,
```

For the others, it might be an idea to introduce a new error variant `DeltUnhandled(git2::Delta)`, just so we can jeep track of one of these popping up.

Reply:
c5cc0a2